bin/brew

```diff
@@ -110,22 +110,11 @@ then
 elif [[ -n "${HOME-}" ]]
 then
   HOMEBREW_USER_CONFIG_HOME="${HOME}/.homebrew"
-elif [[ -n "${USER-}" ]]
+elif [[ -n "${USER-}" ]] || [[ -n "${LOGNAME-}" ]]
 then
-  if [[ "${OSTYPE}" == "darwin"* ]]
-  then
-    HOMEBREW_USER_CONFIG_HOME="/Users/${USER}/.homebrew"
-  else
-    HOMEBREW_USER_CONFIG_HOME="/home/${USER}/.homebrew"
-  fi
-elif [[ -n "${LOGNAME-}" ]]
-then
-  if [[ "${OSTYPE}" == "darwin"* ]]
-  then
-    HOMEBREW_USER_CONFIG_HOME="/Users/${LOGNAME}/.homebrew"
-  else
-    HOMEBREW_USER_CONFIG_HOME="/home/${LOGNAME}/.homebrew"
-  fi
+  # https://superuser.com/a/1613980
+  HOME="$(/bin/bash -c "cd ~$(printf %q "${USER:-${LOGNAME}}") && pwd")"
```

Comment:
Great!

Reply:
@carlocab Have you benchmarked this with `hyperfine`? Given it's shelling out to another `bash`: I strongly suspect it will cause noticeable slowdown for e.g. `brew --prefix`. This is why I avoided this approach before.

Reply:
Before this PR:
```
Benchmark 1: HOME= USER= brew --prefix
  Time (mean ± σ):      54.8 ms ±   7.7 ms    [User: 8.8 ms, System: 10.9 ms]
  Range (min … max):    47.2 ms …  89.7 ms    51 runs
```
After this PR:
```
Benchmark 1: HOME= USER= brew --prefix
  Time (mean ± σ):      60.7 ms ±   4.0 ms    [User: 8.7 ms, System: 10.4 ms]
  Range (min … max):    55.8 ms …  75.3 ms    47 runs
```
It's a little bit slower, but not really noticeably so, I think. That said, this is for the codepath where `HOME` is not set, which is unlikely to affect many users.